src/librustdoc/html/highlight.rs

```diff
@@ -552,8 +558,12 @@ pub(super) fn write_code(
     if let Some(line_info) = line_info {
         token_handler.line = line_info.start_line - 1;
         token_handler.max_lines = line_info.max_lines;
-        if let Some(text) = token_handler.handle_backline() {
-            token_handler.push_token_without_backline_check(None, Cow::Owned(text), false);
+        if let Some(backline) = token_handler.handle_backline() {
+            token_handler.push_token_without_backline_check(
+                None,
+                Cow::Owned(backline.to_string()),
```

Comment:
The fact that we're using `Cow::Owned` here didn't sit right to me, so I did some digging and found out `push_token_without_backline_check` has overly restrictive lifetimes.  It should use `text: Cow<'_, str>`, not `text: Cow<'a, str>`.

Actually, I think it could even be just `text: &str`, since it doesn't look like the code ever takes advantage of the `Owned` case and instead just uses it as `impl Display`, but that would require more refactoring than just changing a lifetime